src/hipscat_cloudtests/temp_cloud_directory.py

```diff
@@ -63,7 +66,10 @@ def close(self, num_retries=4):
             for attempt_number in range(1, num_retries + 1):
                 ## Try
                 try:
-                    file_io.remove_directory(self.temp_path)
+                    if self.real_directories:
+                        file_io.remove_directory(self.temp_path)
+                    else:
+                        _try_remove_contents(self.temp_path)
```

Comment:
`shutil.rmtree` should work?

Reply:
Nope - the paths are s3 paths, and shutil doesn't understand those, so we remove via universal pathlib, which doesn't have a recursive delete mechanism and you just have to walk the directory.